treeherder/client/thclient/client.py

```diff
@@ -552,6 +553,26 @@ def validate(self):
         for d in self.data:
             d.validate()
 
+    def get_chunks(self, chunk_size):
+        """
+        Return a generator of new collections broken into chunks of size ``chunk_size``.
+
+        Each chunk will be a ``TreeherderCollection`` of the same
+        type as the original with a max of ``chunk_size`` count of
+        ``TreeherderData`` objects.
+
+        Each collection must then be POSTed individually.
+        """
+        for i in range(0, len(self.data), chunk_size):
+            # we must copy not only the data chunk,
+            # but also the endpoint_base or any other field of the
+            # collection.  In the case of a TreeherderJobCollection,
+            # this is determined in the constructor.
+
+            chunk = self.__class__(self.data[i:i + chunk_size])
```

Comment:
I can't just pass the ``endpoint_base`` here in the constructor of the child, because it's determined in the constructor of the child class, not the ``TreeherderCollection`` parent class.  So I just need to copy it after creating.